src/Sulu/Component/Rest/ListBuilder/Doctrine/DoctrineListBuilder.php

```diff
@@ -198,7 +198,7 @@ public function execute()
         $this->expressionFields = $this->getUniqueExpressionFieldDescriptors($this->expressions);
 
         if (!$this->limit && empty($this->expressions)) {
-            $queryBuilder = $this->createFullQueryBuilder($this->createQueryBuilder());
+            $queryBuilder = $this->createFullQueryBuilder($this->createQueryBuilder($this->getJoins(true)));
```

Comment:
I would not extend the `getJoins` method. I only would maybe change this `if` statement if this really needed: 

```php
if (!$this->limit && empty($this->expressions) && !$this->search) {
```

Reply:
as I think we are now adding `expressionFields` in a query where no `expressions` are.

Reply:
i think its a bit confusing that the `getJoins()` method does not return the joins for the `this->expressionFields` then. the `getAllFields` method already has a similar `onlyReturnFilterFields` parameter. but on the other hand, the class is already confusing anyway, so i am okay with both ways 🙂

Reply:
I'm just a little bit afraid about add new parameters to an protected function on a not final class. And yes I find the class also very confusing at current state.

Reply:
is okay for me - have removed the parameter 🙂